### PR TITLE
Refine declarative config errors

### PR DESCRIPTION
**Pull request by dhaus67**
## Description

This PR refines the current errors emitted by the configuration unmarshal of declarative config.

Previously, this was a multierror with little value:

```
unable to unmarshal the configuration: 4 errors occurred:
 * yaml: unmarshal errors:
line 1: field cool not found in type declarativeconfig.AuthProvider
line 2: field value not found in type declarativeconfig.AuthProvider
* yaml: unmarshal errors:
line 1: field cool not found in type 
declarativeconfig.AccessScope
line 2: field value not found in type declarativeconfig.AccessScope
* yaml: unmarshal errors:
line 1: field cool not found in type declarativeconfig.PermissionSet
line 2: field value not found in type declarativeconfig.PermissionSet
* yaml: unmarshal errors:
line 1: field cool not found in type declarativeconfig.Role
line 2: field value not found in type declarativeconfig.Role
```

Instead, this error has been now refactor to:
- short-circuit in case `errox.InvalidArgs` is emitted. This error will be emitted in case an invalid value is given.
- simplify the error if neither of the supported configuration types could be unmarshalled.

Additionally, as a small caveat, moved the error logs to `Debugf` instead of `Errorf` for the declarative config manager, as errors may be persistent across multiple reconciliation runs, and hence would create a log statement every 30 seconds (which would be 2880 log lines emitted per day, with little value).

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

- see CI, adjusted tests.

### CODE REVIEW

**Comment by dhaus67** [comment]
Current dependencies on/for this PR:
* master
  * **PR #5847** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/5847" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/5847?utm_source=stack-comment).

File: central/declarativeconfig/manager_impl.go
Comment (ivan-degtiarenko) [inline central/declarativeconfig/manager_impl.go:313]: I would argue that it's OK to switch to `Error` log level for errors which modify integration health - however, for errors that are only displayed in logs switching to `Debug` level would make them essentially invisible 🤔

File: central/declarativeconfig/manager_impl.go
Comment (ivan-degtiarenko) [inline central/declarativeconfig/manager_impl.go:419]: Same concern as above

File: central/declarativeconfig/manager_impl.go
Comment (ivan-degtiarenko) [inline central/declarativeconfig/manager_impl.go:363]: Same concern as above

File: pkg/declarativeconfig/configuration.go
Comment (dhaus67) [reply pkg/declarativeconfig/configuration.go:78]: I made it into a constant, but I'm not sure I like it to be honest. I'll leave it up to you whether you feel this is better.

File: pkg/declarativeconfig/configuration.go
Comment (ivan-degtiarenko) [inline pkg/declarativeconfig/configuration.go:23]: What I thought about here is rather `const map[string]func() Configuration` and `supportedTypes` string is built on `init` out of map keys. Perhaps it's overkill, although thinking about the future to have one place in the file where you need to make changes to add a new type can be beneficial. 

Either way, I'll approve and leave the decision of whether to roll back or change to `const map[string]func() Configuration` to you.

**Review by ivan-degtiarenko — APPROVED** [review]
LGTM modulo discussion around `supportedTypes` string

File: pkg/declarativeconfig/configuration.go
Comment (dhaus67) [reply pkg/declarativeconfig/configuration.go:23]: Hm, I'm not sure it makes a difference though? You'd still need to include a configuration type twice: first you'd have to define and export it, then you have to add it either to an init function or the helper function I had initially. I don't really see any need to introduce the init function here to be honest, but maybe I'm missing something 🤷

**Pull request merged**
